pkg/cmd/account/aws/create/create.go

```diff
@@ -0,0 +1,173 @@
+package create
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/MakeNowJust/heredoc/v2"
+	"github.com/OctopusDeploy/cli/pkg/cmd/account/helper"
+	"github.com/OctopusDeploy/cli/pkg/constants"
+	"github.com/OctopusDeploy/cli/pkg/factory"
+	"github.com/OctopusDeploy/cli/pkg/output"
+	"github.com/OctopusDeploy/cli/pkg/question"
+	"github.com/OctopusDeploy/cli/pkg/question/selectors"
+	"github.com/OctopusDeploy/cli/pkg/surveyext"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/accounts"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/client"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/core"
+	"github.com/briandowns/spinner"
+	"github.com/spf13/cobra"
+)
+
+type CreateOptions struct {
+	Writer  io.Writer
+	Octopus *client.Client
+	Ask     question.Asker
+	Spinner *spinner.Spinner
+
+	Name         string
+	Description  string
+	AccessKey    string
+	SecretKey    string
+	Environments []string
+
+	NoPrompt bool
+}
+
+func NewCmdCreate(f factory.Factory) *cobra.Command {
+	opts := &CreateOptions{
+		Ask:     f.Ask,
+		Spinner: f.Spinner(),
+	}
+	descriptionFilePath := ""
+
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Creates an aws account",
+		Long:  "Creates an aws account in an instance of Octopus Deploy.",
+		Example: fmt.Sprintf(heredoc.Doc(`
+			$ %s account aws create"
+		`), constants.ExecutableName),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := f.GetSpacedClient()
+			if err != nil {
+				return err
+			}
+			opts.Octopus = client
+			opts.Writer = cmd.OutOrStdout()
+			if descriptionFilePath != "" {
+				data, err := os.ReadFile(descriptionFilePath)
+				if err != nil {
+					return err
+				}
+				opts.Description = string(data)
+			}
+			noPrompt, err := cmd.Flags().GetBool(constants.FlagNoPrompt)
```

Comment:
If factory.Ask is nil, it means we're in CI mode; otherwise we're in interactive mode. The decision making (flags, environment variables, etc) is centralised down in the factory. There's an `IsPromptEnabled()` helper method on the factory that you can check for a nice-looking version of the nil check.

Suggest you use that rather than checking flags specifically here